control-plane/agents/src/bin/ha/cluster/switchover.rs

```diff
@@ -354,59 +363,68 @@ impl SwitchOverEngine {
     /// Switchover request to be handled synchronously in each worker task.
     async fn worker(self, recv: Arc<Mutex<UnboundedReceiver<SwitchOverRequest>>>, worker_num: u8) {
         loop {
-            let mut lo = recv.lock().await;
-            if let Some(mut q) = lo.recv().await {
-                drop(lo);
-                info!(
-                    volume.uuid = %q.volume_id,
-                    worker = worker_num,
-                    "Volume switchover picked by worker",
-                );
-                loop {
-                    // TODO: handle error handling properly
-                    let result = match q.stage {
-                        Stage::Init => q.initialize(&self.etcd).await,
-                        Stage::RepublishVolume => q.republish_volume(&self.etcd).await,
-                        Stage::ReplacePath => q.replace_path(&self.etcd, &self.nodes).await,
-                        Stage::DeleteTarget => q.delete_target(&self.etcd).await,
-                        Stage::Errored => match q.errored_switchover(&self.etcd).await {
-                            Ok(_) => break,
-                            Err(e) => Err(e),
-                        },
-                        Stage::Successful => match q.delete_switchover(&self.etcd).await {
-                            Ok(_) => break,
-                            Err(e) => Err(e),
-                        },
-                    };
-                    match result {
-                        Ok(_) => {
-                            // reset retry count back to the start after successfully completing a
-                            // stage.
-                            q.retry_count = 0;
-                        }
-                        Err(_) => {
-                            info!(volume.uuid=%q.volume_id, "Sending failed Switchover request back to the channel");
-                            q.retry_count += 1;
-                            self.enqueue(q);
-                            break;
-                        }
-                    }
+            let request = match recv.lock().await.recv().await {
+                None => break,
```

Comment:
Is there chance of this being None ever? It can happen only when channel is manually closed we dont do it right?